soco/music_services/music_service.py

```diff
@@ -0,0 +1,780 @@
+        pmap_root = XML.fromstring(pmap.content)
+        # Search translations can appear in Category or CustomCategory elements
+        categories = pmap_root.findall(".//SearchCategories/Category")
+        if categories is None:
+            return self._search_prefix_map
+        for cat in categories:
+            self._search_prefix_map[cat.get('id')] = cat.get('mappedId')
+        custom_categories = pmap_root.findall(
+            ".//SearchCategories/CustomCategory")
+        for cat in custom_categories:
+            self._search_prefix_map[cat.get('stringId')] = cat.get('mappedId')
+        return self._search_prefix_map
+
+    @property
+    def available_search_categories(self):
+        """The list of search categories supported by this service.
+
+        May include 'artists', 'albums', 'tracks', 'playlists',
+        'genres', 'stations', 'tags', or others depending on the service
+
+        """
+        # Some services, eg Spotify, support "all", but do not advertise it
+        return self._get_search_prefix_map().keys()
+
+    ########################################################################
+    #                                                                      #
+    #                           SOAP METHODS.                              #
+    #                                                                      #
+    ########################################################################
+
+    #  Looking at various services, we see that the following SOAP methods
+    #  are implemented, but not all in each service. Probably, the
+    #  Capabilities property indicates which features are implemented, but
+    #  it is not clear precisely how. Some of the more common/useful
+    #  features have been wrapped into instance methods, below.
+    #  See generally: http://musicpartners.sonos.com/node/81
+
+    #    createItem(xs:string favorite)
+    #    createTrialAccount(xs:string deviceId)
+    #    deleteItem(xs:string favorite)
+    #    getAccount()
+    #    getExtendedMetadata(xs:string id)
+    #    getExtendedMetadataText(xs:string id, xs:string Type)
+    #    getLastUpdate()
+    #    getMediaMetadata(xs:string id)
+    #    getMediaURI(xs:string id)
+    #    getMetadata(xs:string id, xs:int index, xs:int count,xs:boolean
+    #                recursive)
+    #    getScrollIndices(xs:string id)
+    #    getSessionId(xs:string username, xs:string password)
+    #    mergeTrialccount(xs:string deviceId)
+    #    rateItem(id id, xs:integer rating)
+    #    search(xs:string id, xs:string term, xs:string index, xs:int count)
+    #    setPlayedSeconds(id id, xs:int seconds)
+
+    def get_metadata(
+            self, item_id='root', index=0, count=100, recursive=False):
+        """Get metadata for a container or item.
+
+        Args:
+            item_id (str): The container or item to browse. Defaults to the
+                root item
```

Comment:
Yes, I think this is what is used for browsing. You can start by querying the root (or doing a search). You can then `get_metadata` on albums, which will return tracks, and you can then `get_metadata` on the individual tracks one you know their ids.